_posts/2022-09-07-storage-write-removal-before-conditional-termination.md

```diff
@@ -0,0 +1,133 @@
+---
+layout: post
+published: true
+title: 'Removal of Storage Writes Before Calls to Conditionally Transaction-Terminating Functions'
+date: '2022-09-22'
+author: Solidity Team
+category: Security Alerts
+---
+On September 5, 2022, a bug in Solidity's Yul optimizer was found by differential fuzzing.
+
+The bug was introduced in version 0.8.13 and primarily affects via-IR code generation with enabled optimizer. However, it can also affect optimized legacy code generation with inline assembly blocks involving user-defined functions.
+[Solidity version 0.8.17](https://github.com/ethereum/solidity/releases/tag/v0.8.17), released on September 07, 2022, provides a fix.
+
+We assigned the bug a severity of "medium/high".
+
+
+## Which Contracts are Affected?
+
+A prerequisite for a contract to be affected is the use of ``return(...)`` or ``stop()`` in an inline assembly block (the Solidity ``return`` statement is not affected). Such an inline assembly call will not return (as in continue control flow after the statement), but instead result in an early successful (i.e. non-reverting) termination of the entire transaction.
+
+If this termination of the transaction happens conditionally within a function, i.e. the function can either successfully terminate the transaction or return (as in exit to continue control flow after the call at the call site), the optimizer may incorrectly remove state changes before a call to the function.
```

Comment:
I think it might be better to avoid clarifications here and instead just add a paragraph about the difference between termination and a revert and between Solidity and Yul `return` in technical details.

Reply:
I hope this is clearer in the new version. Generally, a bit unfortunate that we pretty much have to distinguish between returning functions and functions calling ``return(...)``, though...